### DISCUSSION TITLE
Table searchable() is not query correctly

### DISCUSSION THREAD

**Discussion by kengheng1024** [filamentphp/filament#4512] [General]
The searchable in the table is not return the correct results. I've a country table with Code and Name, the Code column is alpha-2 code, both columns has turned on the searchable() method.

1) If I query "US" in the search field, the alpha-2 code won't return "US", but those name contained "US" returned.
2) The search won't query start from the first character, but start from second character, 
eg: query "united", no result return,
query "nited", return United States, United Kingdom, United Arab Emirates and etc... any with "nited"

My env info:
filament: 2.16
php:  8.1.11
laravel: 9.35.1
DB: oracle
Db Driver: yajra/laravel-oci8: 9.0

Suspected it is bug.

**Comment by kengheng1024** [comment]
Query "US", alpha-2 code columm not return result
![Countries_-_Laravel_US](https://user-images.githubusercontent.com/26240069/195535958-cb8ab5d1-e215-434a-84ca-440913a14973.png)

Query "united", no result return
![Countries_-_Laravel_united](https://user-images.githubusercontent.com/26240069/195536010-38948a9e-499b-4588-8a00-b95d8f98066d.png)

Query "nited", return results, query start from second character
![Countries_-_Laravel_nited](https://user-images.githubusercontent.com/26240069/195536055-2b65be42-debe-46ba-a9de-df9324ca59b6.png)

**Comment by danharrin** [comment]
Laravel doesnt have native support for Oracle database, so the search is probably case sensitive. You can PR support to Filament if you want, but I'm not going to because I don't use that database.

**Comment by kengheng1024** [comment]
Hi danharrin,
thanks for your highlight about the case sensitive. I've found a workaround as below:
at InteractsWithTableQuery.php
line 108:
old:
                    fn (Builder $query): Builder => $query->{$whereClause}(
                        $searchColumnName,
                        $searchOperator,
                        "%{$search}%",
                    ),
new:
                    fn (Builder $query): Builder => $query->{"{$whereClause}Raw"}(
                        "lower({$searchColumnName}) {$searchOperator} ?",
                        "%{$search}%",
                    ),

**Comment by Munaziri** [comment]
Hi 👋

**Reply by Umar-gambo** [reply]
Good
